src/mono/mono/metadata/CMakeLists.txt

```diff
@@ -91,8 +91,13 @@ set(metadata_common_sources
     lock-tracer.h
     marshal.c
     marshal.h
+    marshal-legacy.c
```

Comment:
I haven't started looking in detail yet, but I would have expected marshal-legacy.c to go into the ilgen_base_sources group (line 15-22)

Reply:
Ok, I will make that change. I don't think I need a detailed review yet. I put it in draft mode while I work out some things I only saw in CI, but it doesn't seem to let me take your off as a reviewer.